eth2util/keystore/keystore.go

```diff
@@ -121,6 +133,104 @@ func LoadKeys(dir string) ([]tbls.PrivateKey, error) {
 	return resp, nil
 }
 
+// LoadKeysSequential returns all secrets stored in dir/keystore-([0-9]*).json 2335 Keystore files
+// using password stored in dir/keystore-([0-9]*).txt.
+// The keystore files are read based on their index, and the returned slice is sorted accordingly.
+func LoadKeysSequential(dir string) ([]tbls.PrivateKey, error) {
+	return loadFiles(dir, "keystore-*.json", func(files []string) ([]string, error) {
+		newFiles, indices, err := orderByKeystoreNum(files)
+		if err != nil {
+			return nil, err
+		}
+
+		if len(indices) == 0 {
+			return nil, errors.New("empty keystore indices")
+		}
+
+		if indices[0] != 0 {
+			return nil, errors.New("keystore indices must start from zero",
+				z.Int("first_index", indices[0]))
+		}
+
+		for sliceIdx, idx := range indices {
+			if sliceIdx == 0 {
+				continue
+			}
+
+			lastIdx := indices[sliceIdx-1] + 1
+
+			if lastIdx != idx {
+				return nil, errors.New("indices are non sequential",
+					z.Int("expected", lastIdx),
+					z.Int("got", idx))
+			}
+		}
+
+		return newFiles, nil
+	})
+}
+
+// LoadKeys returns all secrets stored in dir/keystore-*.json 2335 Keystore files
+// using password stored in dir/keystore-*.txt.
+func LoadKeys(dir string) ([]tbls.PrivateKey, error) {
+	return loadFiles(dir, "keystore-*.json", nil)
+}
+
+// orderByKeystoreNum orders keystore file names by their index in ascending order.
+func orderByKeystoreNum(files []string) ([]string, []int, error) {
+	prefix := filepath.Dir(files[0])
+
+	extractor := regexp.MustCompile(`keystore-(?:insecure-)?([0-9]+).json`)
+
+	var sortErr error
+
+	idxSet := make(map[int]struct{})
+
+	sort.Slice(files, func(i, j int) bool {
+		first := strings.TrimPrefix(files[i], prefix)
+		second := strings.TrimPrefix(files[j], prefix)
+
+		if !extractor.MatchString(first) || !extractor.MatchString(second) {
+			sortErr = errors.New("keystore filenames do not match expected pattern")
```

Comment:
please split this check into two and add expected pattern and actual filename as error param for improved UX and debugging : 

`errors.New("keystore filenames do not match expected pattern 'keystore-%d.json'", z.Str("filename", first)`